core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -504,16 +510,23 @@ private void registerBuiltinSorts() {
         namedWriteableRegistry.register(SortBuilder.class, FieldSortBuilder.NAME, FieldSortBuilder::new);
     }
 
-    private void registerBuiltinFunctionScoreParsers() {
-        registerFunctionScoreParser(new ScriptScoreFunctionParser());
-        registerFunctionScoreParser(new GaussDecayFunctionParser());
-        registerFunctionScoreParser(new LinearDecayFunctionParser());
-        registerFunctionScoreParser(new ExponentialDecayFunctionParser());
-        registerFunctionScoreParser(new RandomScoreFunctionParser());
-        registerFunctionScoreParser(new FieldValueFactorFunctionParser());
+    private void registerBuiltinScoreFunctionParsers() {
+        registerScoreFunction(ScriptScoreFunctionBuilder.FIELD, ScriptScoreFunctionBuilder::fromXContent, ScriptScoreFunctionBuilder.NAME,
```

Comment:
>  What do you think about passing both name and FUNCTION_NAME_FIELD?

Not sure I follow. Every query has a single name now, all the others are either camel case variants or deprecated names. We kept the convention that we have a NAME string constant everywhere so that we use it for getWriteableName, and also to create the parse field that we need to register the query. Why would we need to pass in the name as well?